src/database/keyvalue.rs

```diff
@@ -82,6 +82,13 @@ macro_rules! impl_batch_operations {
             Ok(())
         }
 
+        fn set_last_sync_time(&mut self, ct: ConfirmationTime) -> Result<(), Error> {
```

Comment:
I tried implementing this but I got stuck because we don't have a way to get the tip hash with the current `Blockchain` trait.

Since I would have to make a breaking change to the trait I'm thinking I might as well split it up like you described during our last call.

I'm not sure if we should hold this PR until that's ready or just go ahead and merge it without block_hash and add that later on.

Reply:
I think merge this now and follow up with a PR that adds `Option<BlockHash>` to the struct. I still think that I we don't want to add this to `BlockTime` but rather a struct like `SyncTime` that contains a `BlockTime`. Also `SyncTime` should probably contain the UTC time that the sync was finished (in this PR).

Reply:
Yeah in my local branch I had a separate structure that contained the `BlockTime` and the block hash.

I will add just the UTC time for now, we'll add the block hash in a following PR

Reply:
You might consider adding a field for it in the sqlite table otherwise we have to do a table update and I'm not sure we have a db upgrade plan yet :S?